### PR TITLE
feat(Styleguide): Adds VScode snippets for new stories GM-137

### PR DESCRIPTION
**Pull request by codecaaron**
### Overview

Adding a few unopinionated story snippets to make creating stories simple.

* Adds vscode snippets for common storybook use cases.
* Ensures vscode file associations.

### Snippets
- `component-story`:  A basic story template with all of the imports and tab targets for standard information
- `toc-story`: Creates a basic index story with no component and a table of contents.
- `canvas-block`: Adds a canvas and story element that renders from args with a tab target for the story name and component.

### Usage
1. In any MDX file start typing the prefix and you should see the autocomplete. 
2. Select the snippet
3. Fill out the tab indexes (in some of these you will fill out multiple things being filled out at the same time).

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: GM-137
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@3.1.1-alpha.c5649c.0<br/>@codecademy/macros@1.1.1-alpha.c5649c.0<br/>@codecademy/styleguide@21.14.1-alpha.c5649c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://600602aa51163982c0e7bf0f--gamut-preview.netlify.app>https://600602aa51163982c0e7bf0f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/600602aa51163982c0e7bf0f>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@3.1.1-alpha.c0bcbd.0<br/>@codecademy/macros@1.1.1-alpha.c0bcbd.0<br/>@codecademy/styleguide@21.14.1-alpha.c0bcbd.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://600604109d0752fdd82e9ad3--gamut-preview.netlify.app>https://600604109d0752fdd82e9ad3--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/600604109d0752fdd82e9ad3>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@3.1.1-alpha.6bdaaf.0<br/>@codecademy/macros@1.1.1-alpha.6bdaaf.0<br/>@codecademy/styleguide@21.14.1-alpha.6bdaaf.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://600626cc9d075230542e9ada--gamut-preview.netlify.app>https://600626cc9d075230542e9ada--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/600626cc9d075230542e9ada>Deploy Logs</a></p>

File: .vscode/stories.code-snippets [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      "",
+      "<PropsTable story=\"$1\"/>"
+    ],
+    "description": "Default Component Story Structure."
+  },
+  "Table of Contents Story": {
+    "prefix": "toc-story",
+    "body": [
+      "import title from '@codecademy/macros/lib/title.macro';",
+      "import { Meta  } from '@storybook/addon-docs/dist/blocks';",
```
Comment (JoshuaKGoldberg) [inline .vscode/stories.code-snippets:37]:
```suggestion
      "import { Meta } from '@storybook/addon-docs/dist/blocks';",
```

**Review by JoshuaKGoldberg — APPROVED** [review]
Great idea!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@3.1.1-alpha.eca9c8.0<br/>@codecademy/macros@1.1.1-alpha.eca9c8.0<br/>@codecademy/styleguide@21.14.1-alpha.eca9c8.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6006f0aa3da5ab1b758f205b--gamut-preview.netlify.app>https://6006f0aa3da5ab1b758f205b--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6006f0aa3da5ab1b758f205b>Deploy Logs</a></p>

**Review by dreamwasp — APPROVED** [review]
very cool ✨

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@3.1.1-alpha.b4b9b1.0<br/>@codecademy/macros@1.1.2-alpha.b4b9b1.0<br/>@codecademy/styleguide@21.14.3-alpha.b4b9b1.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://600726c731a9424cbff58a24--gamut-preview.netlify.app>https://600726c731a9424cbff58a24--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/600726c731a9424cbff58a24>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@3.1.1-alpha.41b292.0<br/>@codecademy/gamut-illustrations@0.4.4-alpha.41b292.0<br/>@codecademy/gamut-labs@7.10.4-alpha.41b292.0<br/>@codecademy/gamut@19.11.0-alpha.41b292.0<br/>@codecademy/macros@1.1.2-alpha.41b292.0<br/>@codecademy/styleguide@21.15.0-alpha.41b292.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@3.1.1-alpha.e2ec03.0<br/>@codecademy/gamut-illustrations@0.4.4-alpha.e2ec03.0<br/>@codecademy/gamut-labs@7.10.4-alpha.e2ec03.0<br/>@codecademy/gamut@19.11.0-alpha.e2ec03.0<br/>@codecademy/macros@1.1.2-alpha.e2ec03.0<br/>@codecademy/styleguide@21.15.0-alpha.e2ec03.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60072b1045d3646736c2707d--gamut-preview.netlify.app>https://60072b1045d3646736c2707d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60072b1045d3646736c2707d>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@21.15.1-alpha.adcb31.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://600732b62fdb755ca7d3113a--gamut-preview.netlify.app>https://600732b62fdb755ca7d3113a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/600732b62fdb755ca7d3113a>Deploy Logs</a></p>

**Review by saghdaey — APPROVED** [review]
cant wait to use!!!!!!! thank u

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.4.5-alpha.a5a8c9.0<br/>@codecademy/gamut-labs@7.10.5-alpha.a5a8c9.0<br/>@codecademy/gamut@19.11.1-alpha.a5a8c9.0<br/>@codecademy/styleguide@21.15.2-alpha.a5a8c9.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://600737e3e8e9736172c875e7--gamut-preview.netlify.app>https://600737e3e8e9736172c875e7--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/600737e3e8e9736172c875e7>Deploy Logs</a></p>

**Pull request merged**